blueprints/route/index.js

```diff
@@ -61,49 +61,53 @@ module.exports = {
     return !isBasic && !isIndex && !isApplication;
   },
 
+  shouldTouchRouter: function(name, options) {
+    var entityTouchesRouter = this.shouldEntityTouchRouter(name);
+    var isDummy = !!options.dummy;
+    var isAddon = !!options.project.isEmberCLIAddon();
+    var isAddonDummyOrApp = (isDummy === isAddon);
+
+    return (entityTouchesRouter && isAddonDummyOrApp && !options.dryRun && !options.inRepoAddon && !options.skipRouter);
+  },
+
   afterInstall: function(options) {
-    var entity  = options.entity;
-
-    if (this.shouldTouchRouter(entity.name) && !options.dryRun && !options.project.isEmberCLIAddon() && !options.inRepoAddon && !options.skipRouter) {
-      addRouteToRouter(entity.name, {
-        root: options.project.root,
-        path: options.path
-      });
-      this.ui.writeLine('updating router');
-      this._writeStatusToUI(chalk.green, 'add route', entity.name);
-    }
+    updateRouter.call(this, 'add', options);
   },
 
   afterUninstall: function(options) {
-    var entity  = options.entity;
+    updateRouter.call(this, 'remove', options);
+  }
+};
 
-    if (this.shouldTouchRouter(entity.name) && !options.dryRun && !options.project.isEmberCLIAddon() && !options.inRepoAddon && !options.skipRouter) {
-      removeRouteFromRouter(entity.name, {
-        root: options.project.root
-      });
+function updateRouter(action, options) {
+  var entity = options.entity;
 
-      this.ui.writeLine('updating router');
-      this._writeStatusToUI(chalk.red, 'remove route', entity.name);
-    }
+  if (this.shouldTouchRouter(entity.name, options)) {
+    writeRoute(action, entity.name, options);
+
+    this.ui.writeLine('updating router');
+    this._writeStatusToUI(chalk.red, action + ' route', entity.name);
```

Comment:
Is there a specific reason we're using red for adding and removing routes? Might be confusing for users.